app/controllers/course/assessment/question/programming_controller.rb

```diff
@@ -58,4 +44,31 @@ def programming_question_params
       skill_ids: []
     )
   end
+
+  def save_and_redirect(template)
+    if @programming_question.save
+      if @programming_question.import_job
+        redirect_to job_path(@programming_question.import_job)
+      else
+        redirect_to course_assessment_path(current_course, @assessment),
+                    success: t('.success')
+      end
+    else
+      render template
```

Comment:
Just to confirm, this would not be needed if later React comes in the frontend ?  Cause render 'template' is for traditional rails.